src/test/java/com/amazon/ion/impl/bin/IonRawBinaryWriterTest_1_1.kt

```diff
@@ -219,4 +244,165 @@ class IonRawBinaryWriterTest_1_1 {
             repeat(8) { stepOut() }
         }
     }
+
+    @Test
+    fun `writeAnnotations with empty int array should write no annotations`() {
+        assertWriterOutputEquals("5E") {
+            writeAnnotations(intArrayOf())
+            writeBool(true)
+        }
+    }
+
+    @Test
+    fun `write one sid annotation`() {
+        val expectedBytes = "E4 07 5E"
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(3)
+            writeBool(true)
+        }
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(3)
+            writeAnnotations(intArrayOf())
+            writeAnnotations(arrayOf<CharSequence>())
+            writeBool(true)
+        }
+    }
+
+    @Test
+    fun `write two sid annotations`() {
+        val expectedBytes = "E5 07 09 5E"
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(3)
+            writeAnnotations(4)
+            writeBool(true)
+        }
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(3, 4)
+            writeBool(true)
+        }
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(intArrayOf(3, 4))
+            writeBool(true)
+        }
+    }
+
+    @Test
+    fun `write three sid annotations`() {
+        val expectedBytes = "E6 09 07 09 02 04 5E"
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(3)
+            writeAnnotations(4)
+            writeAnnotations(256)
+            writeBool(true)
+        }
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(3)
+            writeAnnotations(4, 256)
+            writeBool(true)
+        }
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(intArrayOf(3, 4))
+            writeAnnotations(256)
+            writeBool(true)
+        }
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(intArrayOf(3, 4, 256))
+            writeBool(true)
+        }
+    }
+
+    @Test
+    fun `write one inline annotation`() {
+        val expectedBytes = "E7 FB 66 6F 6F 5F"
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations("foo")
+            writeBool(false)
+        }
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations("foo")
+            writeAnnotations(intArrayOf())
+            writeBool(false)
+        }
+    }
+
+    @Test
+    fun `write two inline annotations`() {
+        val expectedBytes = "E8 FB 66 6F 6F FB 62 61 72 5F"
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations("foo")
+            writeAnnotations("bar")
+            writeBool(false)
+        }
+        assertWriterOutputEquals(expectedBytes) {
+            writeAnnotations(arrayOf("foo", "bar"))
+            writeBool(false)
+        }
+    }
+
+    @Test
+    fun `write three inline annotations`() {
```

Comment:
Looks like a copy/paste error here. These need to be changed to inline text.

Reply:
Good catch. Thanks.